internal/pkg/oasparser/configGenerator.go

```diff
@@ -51,20 +51,28 @@ func GetProductionSources(location string) ([]types.Resource, []types.Resource,
 		endpointsP = append(endpointsP,endpoints...)
 	}
 
-	vHost_NameP := "serviceProd_" + strings.Replace(mgwSwaggers[0].GetTitle(), " ", "", -1) + mgwSwaggers[0].GetVersion()
+	envoyNodeProd := new(envoy.EnvoyNode)
 
-	vHostP, _ := e.CreateVirtualHost(vHost_NameP, routesP)
+	if len(mgwSwaggers) > 0 {
+		vHost_NameP := "serviceProd_" + strings.Replace(mgwSwaggers[0].GetTitle(), " ", "", -1) + mgwSwaggers[0].GetVersion()
 
-	listenerNameP := "listenerProd_1"
-	routeConfigNameP := "routeProd_" + strings.Replace(mgwSwaggers[0].GetTitle(), " ", "", -1) + mgwSwaggers[0].GetVersion()
+		vHostP, _ := e.CreateVirtualHost(vHost_NameP, routesP)
 
-	listnerProd := e.CreateListener(listenerNameP, routeConfigNameP, vHostP)
+		listenerNameP := "listenerProd_1"
+		routeConfigNameP := "routeProd_" + strings.Replace(mgwSwaggers[0].GetTitle(), " ", "", -1) + mgwSwaggers[0].GetVersion()
+
+		listnerProd := e.CreateListener(listenerNameP, routeConfigNameP, vHostP)
+
+
```

Comment:
Remove unnecessary lines. Fix in other places as well.

Reply:
done